src/webgpu/api/validation/copyBufferToBuffer.spec.ts

```diff
@@ -0,0 +1,287 @@
+export const description = `
+copyBufferToBuffer tests.
+
+Test Plan:
+* Buffer is valid/invalid
+  - the source buffer is invalid
+  - the destination buffer is invalid
+* Buffer usages
+  - the source buffer is created without GPUBufferUsage::COPY_SRC
+  - the destination buffer is created without GPUBufferUsage::COPY_DEST
+* CopySize
+  - copySize is not a multiple of 4
+  - copySize is 0
+* copy offsets
+  - sourceOffset is not a multiple of 4
+  - destinationOffset is not a multiple of 4
+* Arthimetic overflow
+  - (sourceOffset + copySize) is overflow
+  - (destinationOffset + copySize) is overflow
+* Out of bounds
+  - (sourceOffset + copySize) > size of source buffer
+  - (destinationOffset + copySize) > size of destination buffer
+* Source buffer and destination buffer are the same buffer
+`;
+
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { kMaxSafeMultipleOf8 } from '../../util/math.js';
+
+import { ValidationTest } from './validation_test.js';
+
+class F extends ValidationTest {
+  TestCopyBufferToBuffer(options: {
+    srcBuffer: GPUBuffer;
+    srcOffset: number;
+    dstBuffer: GPUBuffer;
+    dstOffset: number;
+    copySize: number;
+    isSuccess: boolean;
+  }): void {
+    const { srcBuffer, srcOffset, dstBuffer, dstOffset, copySize, isSuccess } = options;
+
+    const commandEncoder = this.device.createCommandEncoder();
+    commandEncoder.copyBufferToBuffer(srcBuffer, srcOffset, dstBuffer, dstOffset, copySize);
+
+    this.expectValidationError(() => {
+      commandEncoder.finish();
+    }, !isSuccess);
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('copy_with_invalid_buffer').fn(async t => {
+  const bufferSize = 16;
+
+  const validBuffer = t.device.createBuffer({
+    size: bufferSize,
+    usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+  });
+
+  const errorBuffer = t.getErrorBuffer();
+
+  const srcOffset = 0;
```

Comment:
IMHO these constants don't help with readability because when looking at a `TestCopyBufferToBuffer` call we need to look up in the function to know what the parameters are. Maybe they could be inlined in the `TestCopyBufferToBuffer` calls instead. Ditto for other tests.